### PR TITLE
Add APIs for custom registries

### PR DESCRIPTION
**Pull request by praveingk**
This PR addresses an enhancement I created : [Issue 287](https://github.com/vmware/go-ipfix/issues/287)

### CODE REVIEW

File: pkg/registry/registry.go [hunk 1]
```diff
@@ -73,6 +73,26 @@ var (
...
+func InitNewRegistry(customEnterpriseID uint32) error {
+	if globalRegistryByID == nil {
+		return fmt.Errorf("Please use LoadRegistry before registering custom registry")
+	}
+	globalRegistryByID[customEnterpriseID] = make(map[uint16]*entities.InfoElement)
+	globalRegistryByName[customEnterpriseID] = make(map[string]*entities.InfoElement)
+	return nil
+}
+
+func RegisterIE(ie entities.InfoElement, enterpriseID uint32) error {
```
Comment (mariomac) [inline pkg/registry/registry.go:85]: I'd maybe rename the function to something like `RegisterInfoElement` put `PutInfoElement`.

**Review by srikartati — APPROVED** [review]
Looks fine to me. 

I understand that we used globals to have the existing registries as we were working with only 2 registries. In the long term, we should move to struct-based implementation to have single struct per registry. Let user load the registry he wants.

**Comment by praveingk** [comment]
> Looks fine to me.
> 
> I understand that we used globals to have the existing registries as we were working with only 2 registries. In the long term, we should move to struct-based implementation to have single struct per registry. Let user load the registry he wants.

Thanks @srikartati , I had just renamed RegisterIE to PutInfoElement, Can you kindly check and approve again, if its ok.

**Review by dreamtalen — APPROVED** [review]
LGTM

**Comment by yanjunz97** [comment]
Thanks for the contribution! Overall LGTM, could you fix the error here: https://github.com/vmware/go-ipfix/actions/runs/3571268704/jobs/6015512732 by running `make codegen`?

**Comment by yanjunz97** [comment]
The CIs are fixed now. We can merge this PR after you rebase it to main to pass all the checks.

**Comment by praveingk** [comment]
Thanks @yanjunz97 and @dreamtalen , I have rebased the branch to main.

**Comment by praveingk** [comment]
Thanks for merging @yanjunz97, @dreamtalen ,  Can you please create a release tag of the latest version, which I can use in my project to consume the APIs.

**Pull request merged**